packages/react-native-skia/integratedApp/webSocketHelper.js

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed URLr the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+
+//const fs = require('fs');
+const React = require('react');
+const ReactNative = require('react-native');
+const {View, Text} = ReactNative;
+const {TestModule} = ReactNative.NativeModules;
+import Config from './config.json'
+
+let resolution = Config.size.low;
+let fontSize1 = resolution.textStyle.fontSize;
+
+const multi_url = 'ws://localhost:9898';
+let TextSent = <Text />
+
+let bgColor = '';
+
+type State = {
+	url1: string,
+	fetchStatus1: ?string,
+	socket1: ?WebSocket,
+	socketState1: ?number,
+	lastSocketEvent1: ?string,
+	lastMessage1: ?string | ?ArrayBuffer,
+	testMessage1: string,
+	testExpectedResponse1: string,
+	testFlag: ?number,
+	...
+};
+
+class WebSocketTest extends React.Component<{...}, State> {
+
+	state: State = {
+	url1: multi_url,
+	fetchStatus1: null,
+	socket1: null,
+	socketState1: null,
+	lastSocketEvent1: null,
+	lastMessage1: null,
+	testMessage1: 'testMessage1',
+	testExpectedResponse1: "",
+	testFlag: 1,
+	};
+
+	_waitFor = (condition: any, timeout: any, callback: any) => {
+		let remaining = timeout;
+		const timeoutFunction = function() {
+			if (condition()) {
+				callback(true);
+				return;
+			}
+			remaining--;
+			if (remaining === 0) {
+				callback(false);
+			} else {
+				setTimeout(timeoutFunction, 1000);
+			}
+		};
+		setTimeout(timeoutFunction, 1000);
+	};
+
+	_connect1 = () => {
+		console.log("URL: ", this.state.url1);
+		const socket1 = new WebSocket(this.state.url1);
+		console.log("Established new websocket",socket1);
+		socket1.addEventListener("open", this._onSocketOpenEvent1);
+		socket1.addEventListener("message", this._onSocketMessageEvent1);
+		socket1.addEventListener("close", this._onSocketCloseEvent1);

```

Comment:
would be good to handle the exceptions too. :-)

Reply:
Done